lib/commons/dom/is-visible.js

```diff
@@ -39,6 +39,39 @@ function isClipped(style) {
 	return false;
 }
 
+/**
+ * Check `AREA` element is visible
+ * - validate if it is a child of `map`
+ * - ensure `map` is referred by `img` using the `usemap` attribute
+ * @param {Element} areaEl `AREA` element
+ * @retruns {Boolean}
+ */
+function isAreaVisible(el, screenReader, recursed) {
+	/**
+	 * Note:
+	 * - Verified that `map` element cannot refer to `area` elements across different document trees
+	 * - Verified that `map` element does not get affected by altering `display` property
+	 * - using `.closest` fails in `phantomjs`, hence using `dom.findUp`
+	 */
+	const mapEl = dom.findUp(el, 'map');
+	if (!mapEl) {
+		return false;
+	}
+
+	const mapElName = mapEl.getAttribute('name');
+	if (!mapElName) {
+		return false;
+	}
+	const refs = document.querySelectorAll(
```

Comment:
This is the wrong way around. This checks that the `img` is in the light DOM, instead of checking that the `map` is in the light DOM. The `img` can be anywhere in the tree. You should add some shadow DOM tests for this to prevent regression.